docs/containers-transports.5.md

```diff
@@ -0,0 +1,109 @@
+% CONTAINERS-TRANSPORTS(5) Containers Transports Man Page
+% Valentin Rothberg
+% April 2019
+
+## NAME
+
+containers-transports - description of supported transports for copying and storing container images
+
+## DESCRIPTION
+
+Tools which use the containers/image library, including skopeo(1), buildah(1), podman(1), all share a common syntax for referring to container images in various locations.
+The general form of the syntax is _transport:details_, where details are dependent on the specified transport, which are documented below.
+
+### **containers-storage:** [storage-specifier]{image-id|docker-reference[@image-id]}
+
+An image located in a local containers storage.
+The format of _docker-reference_ is described in detail in the **docker** transport.
+
+The _storage-specifier_ allows for referencing storage locations on the file system and has the format `[[driver@]root[+run-root][:options]]` where the optional `driver` refers to the storage driver (e.g., overlay or btrfs) and where `root` is an absolute path to the storage's root directory.
+`run-root` allows to further specify the run directory of the storage where all temporary writable content is stored.
+The optional `options` are a comma-separated list of driver-specific options.
+Please refer to containers-storage.conf(5) for further information on the drivers and supported options.
+
+### **dir:**_path_
+
+An existing local directory _path_ storing the manifest, layer tarballs and signatures as individual files.
+This is a non-standardized format, primarily useful for debugging or noninvasive container inspection.
+
+### **docker://**_docker-reference_
+
+An image in a registry implementing the "Docker Registry HTTP API V2".
+By default, uses the authorization state in `$XDG_RUNTIME_DIR/containers/auth.json`, which is set using podman-login(1).
+If the authorization state is not found there, `$HOME/.docker/config.json` is checked, which is set using docker-login(1).
+The containers-registries.conf(5) further allows for configuring various settings of a registry.
+
+Note that a _docker-reference_ has the following format: `name[:tag|@digest]`.
+While the docker transport does not support both, a tag and a digest, at the same time, some formats, like containers-storage, do.
+Digests can also be used in an image destination as long as the manifest matches the provided digest.
+The digest of images can be explored with skopeo-inspect(1).
+If name does not contain a slash, it is treated as `docker.io/library/name`.
+Otherwise, the component before the first slash is checked whether it is recognized as a `hostname[:port]` (it contains either a . or a :, or the component is exactly localhost).
```

Comment:
When you have the word "whether" it's used to contrast two thoughts and the word "or" is expected later in the sentence and here the 'or' is inside the parens.  Suggest: "Otherwise, the component before the first slash is checked whether it is recognized as a `hostname[:port]` (it contains either a period (.) or a semi-colon (:)), or the component is exactly localhost.

Reply:
The condition being checked here is whether “_name_ is recognized…” (which includes `.`, `:`, and `localhost`), [or whether it is “not recognized” because it does not match that rule].

If I understand you correctly, moving the `localhost` case outside of the parenthesis would mean something else (treating `.`/`:` as one case, `localhost` as the other one, and being silent about other values).

Maybe keep the original text, and `s/whether/if`; I’m not at all sure.